cpp/src/barretenberg/plonk/proof_system/types/proof.hpp

```diff
@@ -10,9 +10,10 @@ namespace proof_system::plonk {
 
 struct proof {
     std::vector<uint8_t> proof_data;
+    // For serialization, serialize as a buffer alias
     void msgpack_pack(auto& packer) const { packer.pack(proof_data); }
     void msgpack_unpack(auto object) { proof_data = (std::vector<uint8_t>)object; }
-
+    void msgpack_schema(auto& packer) const { packer.pack_alias("Proof", "bin32"); }
```

Comment:
not familiar with this code, so forgive my ignorance, but why has this function gone to 
`void msgpack_schema` whilst the diff from `fr.hpp` goes to a `static constexpr char schema_name[] = "fr"`?

Reply:
It's a fairly meaningful refactor. Now, template types that are parameters have a schema_name, while other things have a schema method. It is meant to be cleaner